pytest-embedded/pytest_embedded/dut_factory.py

```diff
@@ -0,0 +1,725 @@
+        erase_nvs=None,
+        skip_check_coredump=None,
+        panic_output_decode_script=None,
+        openocd_prog_path=None,
+        openocd_cli_args=None,
+        gdb_prog_path=None,
+        gdb_cli_args=None,
+        no_gdb=None,
+        qemu_image_path=None,
+        qemu_prog_path=None,
+        qemu_cli_args=None,
+        qemu_extra_args=None,
+        wokwi_cli_path=None,
+        wokwi_timeout=0,
+        skip_regenerate_image=None,
+        encrypt=None,
+        keyfile=None,
+    ):
+        """
+        Create a Device Under Test (DUT) object with customizable parameters.
+
+        Parameters:
+        - embedded_services (str): Comma-separated list of embedded services.
+        - app_path (str): Path to the application.
+        - build_dir (str): Directory for build output (default is 'build').
+        - port (str): Port configuration.
+        - port_location (str): Port location.
+        - port_mac (str): Port MAC address.
+        - target (str): Target configuration.
+        - beta_target (str): Beta target configuration.
+        - baud (int): Baud rate.
+        - skip_autoflash (bool): Skip autoflash flag.
+        - erase_all (bool): Erase all flag.
+        - esptool_baud: ESP tool baud rate.
+        - esp_flash_force (bool): ESP flash force flag.
+        - part_tool: Part tool configuration.
+        - confirm_target_elf_sha256: Confirm target ELF SHA256.
+        - erase_nvs (bool): Erase NVS flag.
+        - skip_check_coredump (bool): Skip coredump check flag.
+        - panic_output_decode_script: Panic output decode script.
+        - openocd_prog_path: OpenOCD program path.
+        - openocd_cli_args: OpenOCD CLI arguments.
+        - gdb_prog_path: GDB program path.
+        - gdb_cli_args: GDB CLI arguments.
+        - no_gdb (bool): No GDB flag.
+        - qemu_image_path: QEMU image path.
+        - qemu_prog_path: QEMU program path.
+        - qemu_cli_args: QEMU CLI arguments.
+        - qemu_extra_args: Additional QEMU arguments.
+        - wokwi_cli_path: Wokwi CLI path.
+        - wokwi_timeout (int): Wokwi timeout.
+        - skip_regenerate_image (bool): Skip image regeneration flag.
+        - encrypt (bool): Encryption flag.
+        - keyfile: Keyfile for encryption.
+
+        Returns:
+        - DUT object: The created Device Under Test object.
+
+        Examples:
+        >>> foo = DutFactory.create(embedded_services='idf,esp', app_path='path_to_hello_world')
+        >>> foo.expect_exact('Hello world!')
+        """
+        layout = []
+        try:
+            global parametrized_fixtures_cache
+            msg_queue = msg_queue_gn()
+            layout.append(msg_queue)
+
+            _pexpect_logfile = os.path.join(
+                parametrized_fixtures_cache['_meta'].logdir, f'custom-dut-{dut_global_index}.txt'
+            )
+            logging.warning('You can get your custom DUT log file at the following path: %s.', _pexpect_logfile)
```

Comment:
```suggestion
            logging.debug('You can get your custom DUT log file at the following path: %s.', _pexpect_logfile)
```

either debug or info is fine, but not a warning.